src/nunavut/lang/c/templates/_composite_type.j2

```diff
@@ -0,0 +1,67 @@
+{%- for constant in composite_type.constants %}
+#define {{ composite_type | c.macrofy }}_{{ constant.name | c.macrofy }} {{ constant | constant_value }}
+{%- endfor %}
+{%- if composite_type.fields | length == 0 %}
+{#- 
+    C++ doesn't tolerate empty types. To maintain compatibility we define any empty
```

Comment:
I'll have to dig it up but I got C++ compatibility warnings when I tried to compile empty structs.

Reply:
Found it:
```
error: empty struct has size 0 in C, size 1 in C++ [-Werror=c++-compat]
```

i.e. -> https://godbolt.org/z/6bLUgC

Reply:
Ah indeed. I think we should keep things consistent and ensure that every generated type is a struct, otherwise, in C++ templated code we might run into complicated corner cases where one might want to use type traits.

Maybe like this?

    typedef struct {{ composite_type | full_reference_name }} { uint8_t _dummy_; } {{ composite_type | full_reference_name }};

Reply:
sure.